code/mapping/ext_modules/mapping_common/transform.py

```diff
@@ -0,0 +1,305 @@
+
+
+@dataclass(init=False, eq=False)
+class Point2(_Coord2):
+    """2 dimensional point.
+
+    Receives both rotation and translation when transformed with a Transform2D"""
+
+    def vector(self) -> "Vector2":
+        return Vector2(self._matrix)
+
+    @staticmethod
+    def new(x: float, y: float) -> "Point2":
+        m = np.array([x, y, 1.0], dtype=np.float64)
+        return Point2(m)
+
+    @staticmethod
+    def zero() -> "Point2":
+        return Point2.new(0.0, 0.0)
+
+    @staticmethod
+    def from_vector(v: "Vector2") -> "Point2":
+        return Point2(v._matrix)
+
+    @staticmethod
+    def from_ros_msg(m: geometry_msgs.Point) -> "Point2":
+        return Point2.new(m.x, m.y)
+
+    def to_ros_msg(self) -> geometry_msgs.Point:
+        return geometry_msgs.Point(x=self.x(), y=self.y(), z=0.0)
+
+    def __add__(self, other):
+        if isinstance(other, Vector2):
+            matrix = self._matrix + other._matrix
+            matrix[2] = 1.0
+            return Point2(matrix)
+        raise TypeError(
+            f"Unsupported operand types for *: '{type(self)}' and '{type(other)}'"
+        )
+
+
+@dataclass(init=False, eq=False)
+class Vector2(_Coord2):
+    """2 dimensional direction vector.
+
+    Receives only the rotation when transformed with a Transform2D"""
+
+    def length(self) -> float:
+        return math.sqrt(self._matrix[0] ** 2 + self._matrix[1] ** 2)
+
+    def normalized(self) -> "Vector2":
+        """Returns this direction Vector with length 1.0
+
+        Returns:
+            Vector2: Vector with length 1.0
+        """
+        return Vector2(self._matrix / self.length())
+
+    def angle_to(self, other: "Vector2") -> float:
+        """Calculates the angle to *other*
+
+        Args:
+            other (Vector2): _description_
+
+        Returns:
+            float: signed angle in radians. Always in interval [-pi,pi].
+
+            - angle > 0: CCW
+            - angle < 0: CW
+        """
+        # Based on https://stackoverflow.com/questions/21483999/
+        # using-atan2-to-find-angle-between-two-vectors/21486462#21486462
+        cross = np.cross(self._matrix[:2], other._matrix[:2])
+        dot = np.dot(self._matrix[:2], other._matrix[:2])
+        return math.atan2(cross, dot)
+
+    def point(self) -> Point2:
+        return Point2(self._matrix)
+
+    @staticmethod
+    def new(x: float, y: float) -> "Vector2":
+        m = np.array([x, y, 1.0], dtype=np.float64)
+        return Vector2(m)
+
+    @staticmethod
+    def zero() -> "Vector2":
+        return Vector2.new(0.0, 0.0)
+
+    @staticmethod
+    def from_point(p: Point2) -> "Vector2":
+        return Vector2(p._matrix)
+
+    @staticmethod
+    def from_ros_msg(m: geometry_msgs.Vector3) -> "Vector2":
+        return Vector2.new(m.x, m.y)
+
+    def to_ros_msg(self) -> geometry_msgs.Vector3:
+        return geometry_msgs.Vector3(x=self.x(), y=self.y(), z=0.0)
+
+    def __mul__(self, other):
+        if isinstance(other, float):
```

Comment:
Good idea.

I will use `isinstance(other, (float, int))` instead, because I always avoid direct type comparsions, especially when cython brings its own number types.